tests/Public/New-GuestConfigurationPolicy.Tests.ps1

```diff
@@ -690,6 +690,75 @@ Describe 'New-GuestConfigurationPolicy' {
+                        {
+                            $deploymentResource.properties.guestConfiguration.contentManagedIdentity | Should -Be 'myManagedIdentity'
+                        }
+                    }
+                }
+
+                Context 'Optional identity parameters - ManagedIdentityResourceId and LocalContentPath without ExcludeArcMachines' {
+                    BeforeAll {
+                        $fileName = $ContentUri.Split('/')[-1]
+                        $filePath = Join-Path -Path $defaultDefinitionsPath -ChildPath $fileName
+                        Invoke-WebRequest $ContentUri -OutFile $filePath
+
+                        $basePolicyParameters['ManagedIdentityResourceId'] = 'myManagedIdentity'
+                        $basePolicyParameters['LocalContentPath'] = $filePath
+                    }
+
+                    It 'Should throw a exception if ExcludeArcMachines is not specified' {
+                        { New-GuestConfigurationPolicy @basePolicyParameters } | Should -Throw -ExpectedMessage 'Custom policies that reference User Assigned Identities will not include Arc enabled servers in the definition. Use -ExcludeArcMachines to continue.'
+                    }
+                }
+
+                Context 'Optional identity parameters - ManagedIdentityResourceId without LocalContentPath' {
+                    BeforeAll {
+                        $basePolicyParameters['ManagedIdentityResourceId'] = 'myManagedIdentity'
+                        $basePolicyParameters['LocalContentPath'] = $null
+                    }
+
+                    It 'Should throw a missing parameter exception if one of the parameters is missing' {
+                        { New-GuestConfigurationPolicy @basePolicyParameters } | Should -Throw -ExpectedMessage 'Both ManagedIdentityResourceId and LocalContentPath must be provided together.'
+                    }
+                }
+
+                Context 'Optional identity parameters - LocalContentPath without ManagedIdentityResourceId' {
+                    BeforeAll {
+                        $fileName = $ContentUri.Split('/')[-1]
+                        $filePath = Join-Path -Path $defaultDefinitionsPath -ChildPath $fileName
+                        Invoke-WebRequest $ContentUri -OutFile $filePath
+
+                        $basePolicyParameters['LocalContentPath'] = $filePath
+                        $basePolicyParameters['ManagedIdentityResourceId'] = $null
+
+                        Remove-Item -Path $filePath
+                    }
+
+                    It 'Should throw a missing parameter exception if one of the parameters is missing' {
+                        { New-GuestConfigurationPolicy @basePolicyParameters } | Should -Throw -ExpectedMessage 'Both ManagedIdentityResourceId and LocalContentPath must be provided together.'
+                    }
+                }
```

Comment:
Can you add a negative test case or update an existing test? If optional identity parameters are NOT provided, we should NOT see contentManagedIdentity in the policy.

Also, this is a nitpick and is subjective but would be cleaner to have one Context block with an It block for each test case instead of a Context block for each test case. And BeforeAll isn't needed here.

Reply:
Added a negative testcase for this - If optional identity parameters are NOT provided, we should NOT see contentManagedIdentity in the policy.
Put all the tests in one context block.